common/tmx_level.go

```diff
@@ -87,15 +84,10 @@ func (t ByFirstgid) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 func (t ByFirstgid) Less(i, j int) bool { return t[i].Firstgid < t[j].Firstgid }
 
 // MUST BE base64 ENCODED and COMPRESSED WITH zlib!
-func createLevelFromTmx(r Resource) (*Level, error) {
+func createLevelFromTmx(tmx string) (*Level, error) {
 	tlvl := &TMXLevel{}
 	lvl := &Level{}
 
-	tmx, err := readTmx(r.url)
-	if err != nil {
-		return lvl, err
-	}
-
 	if err := xml.Unmarshal([]byte(tmx), &tlvl); err != nil {
 		fmt.Printf("Error unmarshalling XML: %v", err)
```

Comment:
Lines 105 and 113 should be fixed accordingly. (and also according to https://github.com/EngoEngine/engo/pull/315/files#r63825301 )

Comment on lines 120 through 125 (now we're at it):

```go
err = binary.Read(zlr, binary.LittleEndian, &nextInt)
if err != nil {
    // this is generally EOF
    break
}
```

I like the part where it's *generally* EOF, but let's check to be sure? If EOF, then break, else return error.